content/articles/boosting-algorithms-part-1/index.md

```diff
@@ -0,0 +1,251 @@
+
+[Ensemble methods](https://www.section.io/engineering-education/ensemble-learning/) are techniques that use multiple models and combine them into one for enhanced results. The term "models" could refer to any model - regression, support vector machines, and kNNs, and the model whose performance has to be improved is called the base model. Although the technique "boosting" uses decision trees to improve the model's accuracy, it can be applied to any base model.
+
+However, it has been observed that boosting a decision tree based model provides better results than boosting other models. One possible explanation could be the structural similarities of the base model (decision tree) and the boosting algorithm.
+
+Bagging and boosting both use an arbitrary `N` number of learners by generating additional data while training. These `N` learners are used to create `M` new training sets by sampling random sets from the original set. The idea behind bagging is to combine the results of the `M` models that are generated from the sampled sets. The models run in parallel and are independent of each other, and the final results are obtained from combining the results of all the models. 
+
+![Bagging](/engineering-education/boosting-algorithms-part-1/bagging.png)
+
+*Source: [Towards Data Science](https://towardsdatascience.com/ensemble-methods-bagging-boosting-and-stacking-c9214a10a205)*
+
+Whereas in boosting, the `M` models are trained sequentially, with the models carrying forth the performance of the previous model in an attempt to correct the errors of the previous model. The performance of the models is analyzed by taking the weighted mean of the performances of the individual model, with weights being assigned by their performance.
+
+![Boosting](/engineering-education/boosting-algorithms-part-1/boosting.png)
+
+*Source: [Towards Data Science](https://towardsdatascience.com/ensemble-methods-bagging-boosting-and-stacking-c9214a10a205)*
+
+### Data Preprocessing
+The article uses the [UCI Machine Learning Mushroom Dataset](https://www.kaggle.com/uciml/mushroom-classification) to implement the AdaBoost and XGBoost algorithms. For the set of features in the dataset, the task is to identify whether the type of mushroom is poisonous or edible. The reader is encouraged to download the dataset and follow along with the code blocks in the article for better understanding.
+
+First, let's import the required libraries.
+
+```py
+import numpy as np
+import pandas as pd
+from sklearn.ensemble import AdaBoostClassifier
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import LabelEncoder
+from sklearn.preprocessing import StandardScaler
+```
+
+Now, let's import the dataset using the `read_csv()` method in Pandas and analyze the number of distinct categories in each feature. If a feature has only one unique value, we can drop it, as it has no significance while building the model.
```

Comment:
You could insert an image containing the dataframe's head() (Top 5 rows) of the dataset